flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecTemporalSort.java

```diff
@@ -0,0 +1,175 @@
+            boolean[] nullIsLast,
+            ExecEdge inputEdges,
+            LogicalType outputType,
+            String description) {
+        super(Collections.singletonList(inputEdges), outputType, description);
+        this.keys = keys;
+        this.orders = orders;
+        this.nullsIsLast = nullIsLast;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected Transformation<RowData> translateToPlanInternal(PlannerBase planner) {
+        TableConfig config = planner.getTableConfig();
+        RowType inputType = (RowType) getInputNodes().get(0).getOutputType();
+        // time ordering needs to be ascending
+        if (orders.length == 0 || !orders[0]) {
+            throw new TableException(
+                    "Sort: Primary sort order of a streaming table must be ascending on time.\n"
+                            + "please re-check sort statement according to the description above");
+        }
+        LogicalType timeType = inputType.getTypeAt(keys[0]);
+        Transformation<RowData> input =
+                (Transformation<RowData>) getInputNodes().get(0).translateToPlan(planner);
+
+        if (timeType instanceof TimestampType) {
+            TimestampType keyType = (TimestampType) timeType;
+            if (keyType.getKind() == TimestampKind.ROWTIME) {
+                return createSortRowTime(inputType, input, config);
+            } else if (keyType.getKind() == TimestampKind.PROCTIME) {
+                return createSortProcTime(inputType, input, config);
+            } else {
+                throw new TableException(
+                        "Sort: Internal Error\n"
+                                + "Normally, this happens unlikely. please contact customer support for this");
+            }
+        } else {
+            throw new TableException(
+                    "Sort: Internal Error\n"
+                            + "Normally, this happens unlikely. please contact customer support for this");
+        }
+    }
+
+    /** Create Sort logic based on processing time. */
+    private Transformation<RowData> createSortProcTime(
+            RowType inputType, Transformation<RowData> input, TableConfig tableConfig) {
+        // if the order has secondary sorting fields in addition to the proctime
+        if (keys.length > 1) {
+            // sort code gen
+            LogicalType[] sortKeyTypes =
+                    IntStream.of(keys).mapToObj(inputType::getTypeAt).toArray(LogicalType[]::new);
+            GeneratedRecordComparator rowComparator =
+                    ComparatorCodeGenerator.gen(
+                            tableConfig,
+                            "ProcTimeSortComparator",
+                            Arrays.copyOfRange(keys, 1, keys.length),
+                            Arrays.copyOfRange(sortKeyTypes, 1, keys.length),
+                            Arrays.copyOfRange(orders, 1, keys.length),
+                            Arrays.copyOfRange(nullsIsLast, 1, keys.length));
```

Comment:
add some comments to explain these code ?

There is one comment `strip off time collation` to explain it in old scala class